pkg/daemon/ceph/osd/volume.go

```diff
@@ -641,6 +641,29 @@ func (a *OsdAgent) initializeDevicesLVMMode(context *clusterd.Context, devices *
 				if device.Config.MetadataDevice != "" {
 					md = device.Config.MetadataDevice
 				}
+				var metadataDevice *sys.LocalDisk
+			OUTER:
+				for _, localDevice := range context.Devices {
+					if localDevice.Name == md || filepath.Join("/dev", localDevice.Name) == md {
+						logger.Infof("%s found in the desired metadata devices", localDevice.Name)
+						metadataDevice = localDevice
+						break
+					} else if strings.HasPrefix(md, "/dev/") {
+						devLinks := strings.Split(localDevice.DevLinks, " ")
+						for _, link := range devLinks {
+							if link == md {
+								logger.Infof("%s found in the desired metadata device (matched by link: %s)", localDevice.Name, link)
+								metadataDevice = localDevice
+								break OUTER
```

Comment:
Ig `OUTER` here symbolizes that we need to re-iterate,

I am not sure but this is considered a not a best practice for writing a code(as we never use `goto-statement`),
I am not sure what you finally trying to achieve but if you describe it I can surely say I can suggest you something better.

Maybe @travisn @BlaineEXE @subhamkrai you should also have a look.

Reply:
Thanks for you suggestion. If we find the corresponding metadata disk, we can break out from the inner loop.

Reply:
Let's not use `go-to` statements like this. It's hard to follow the code, and we have avoided it so far. The better pattern is to make the inner for loop a function of its own. If that function finds a match, then we can just do `break`.